dGame/dComponents/DestroyableComponent.cpp

```diff
@@ -533,19 +532,12 @@ bool DestroyableComponent::CheckValidity(const LWOOBJID target, const bool ignor
 		return true;
 	}
 
-	auto enemyList = GetEnemyFactionsIDs();
-
-	auto candidateList = destroyable->GetFactionIDs();
-
-	for (auto value : candidateList)
-	{
-		if (std::find(enemyList.begin(), enemyList.end(), value) != enemyList.end())
-		{
-			return true;
-		}
-	}
+	// Get if the target entity is an enemy
+	bool isEnemy = IsEnemy(targetEntity);
 
-	return false;
+	// Return true if the target type matches what we are targeting
+	// Friends are entities who are not enemies
```

Comment:
This isn't actually true. There is a separate column in the Factions table that we currently don't use: `friendList`

I'm not certain about the semantics, but I'd suggest you call `DestroyableComponent::IsFriend` as well and we can fix its implementation in a follow up PR.

Reply:
Okay I see. Originally I just had the one call to IsEnemy since also calling IsFriend takes a bit more time, but since IsFriend is planned to be changed in the future, I have updated the code to determine IsEnemy and IsFriend separately.